### PR TITLE
feat(accessibility): snapshot the accessibility tree

### LINKED ISSUE #2033: Feature request: Inspect screen reader output
In order to help facilitate testing of a web app's accessibility, it would be great if we could inspect the screen reader output of a headless browser.

### PR DESCRIPTION
**Pull request by JoelEinbinder**
fixes #2033

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -3431,3 +3452,4 @@ TimeoutError is emitted whenever certain operations are terminated due to timeou
 [UnixTime]: https://en.wikipedia.org/wiki/Unix_time "Unix Time"
 [SecurityDetails]: #class-securitydetails "SecurityDetails"
 [Worker]: #class-worker "Worker"
+[Mouse]: #class-accessibility "Accessibility"
```
Comment (vsemozhetbyt) [inline docs/api.md]:
```suggestion
[Accessibility]: #class-accessibility "Accessibility"
```

File: docs/api.md [hunk 2]
```diff
@@ -1982,6 +1988,21 @@ Shortcut for [(await worker.executionContext()).evaluateHandle(pageFunction, ...
 #### worker.url()
 - returns: <[string]>
 
+### class: Accessibility
+
+The Accessibility class provides methods for inspecting Chromium's accessibility tree. The accessibility tree is used by assistive technology such as a [screen readers](https://en.wikipedia.org/wiki/Screen_reader).
```
Comment (vsemozhetbyt) [inline docs/api.md]:
```suggestion
The Accessibility class provides methods for inspecting Chromium's accessibility tree. The accessibility tree is used by assistive technology such as [screen readers](https://en.wikipedia.org/wiki/Screen_reader).
```

File: docs/api.md (hunk 1 above)
Comment (JoelEinbinder) [reply docs/api.md]: Thanks!

**Comment by vsemozhetbyt** [comment]
FWIW, I confirm my consent)

File: lib/Accessibility.js [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+    }
+
+    return this._isLeafNode() && !!this.name();
+  }
+
+  toString() {
+    // - [${this._children.map(child => child.toString()).join(', ')}]
+    const parts = [];
+    parts.push(`[${this._payload.role.value}]`);
+    // if (this.name())
```
Comment (aslushnikov) [inline lib/Accessibility.js:246]:
```suggestion
```

File: docs/api.md [hunk 4]
```diff
@@ -1982,6 +1988,21 @@ Shortcut for [(await worker.executionContext()).evaluateHandle(pageFunction, ...
 #### worker.url()
 - returns: <[string]>
 
+### class: Accessibility
+
+The Accessibility class provides methods for inspecting Chromium's accessibility tree. The accessibility tree is used by assistive technology such as [screen readers](https://en.wikipedia.org/wiki/Screen_reader).
+
+
+#### accessibility.snapshot()
+- returns: <[Promise]<[string]>>
```
Comment (aslushnikov) [inline docs/api.md:1997]: let's have a structured format here

**Comment by JoelEinbinder** [comment]
Closing this in favor of a new shiny PR.

**Pull request closed without merging**